libtransmission/rpc-server.cc

```diff
@@ -1183,42 +1196,36 @@ tr_rpc_server::tr_rpc_server(tr_session* session_in, tr_variant* settings)
     if (!tr_variantDictFindStrView(settings, key, &sv))
     {
         missing_settings_key(key);
-        address = tr_rpc_inaddr_any;
+        bindAddress->inaddr_any();
     }
     else
     {
-        if (!tr_rpc_address_from_string(&address, sv))
+        if (!tr_rpc_address_from_string(*bindAddress, sv))
         {
             tr_logAddNamedError(MyName, _("%" TR_PRIsv " is not a valid address, falling back to 0.0.0.0."), TR_PRIsv_ARG(sv));
-            address = tr_rpc_inaddr_any;
+            bindAddress->inaddr_any();
         }
-        else if (address.type != TR_RPC_AF_INET && address.type != TR_RPC_AF_INET6 && address.type != TR_RPC_AF_UNIX)
+        else if (!tr_rpc_address_is_valid(*bindAddress))
```

Comment:
Going back to it again, do we really need this check? If address type is not supported, we won't reach it because the block above will be executed instead.